From 8a873e5ccb23d17b02f5059385c44531fa56072d Mon Sep 17 00:00:00 2001
From: Tianshu Qiu <tian.shu.qiu@intel.com>
Date: Wed, 29 Apr 2015 15:31:37 +0800
Subject: [PATCH 369/462] ov2722: Fix KW issue

Change-Id: Ib631ead1ee44cb86bd97a03b5c68687743fa65bd
Fixes-Issue: #JIMINAN-34512
Signed-off-by: Tianshu Qiu <tian.shu.qiu@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/53517
Validation-Android: Wei Tao <wei.tao@intel.com>
Approver: Bin B Han <bin.b.han@intel.com>
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Build: IMIN (SC)
Integrator: Wei Tao <wei.tao@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 drivers/external_drivers/camera/drivers/media/i2c/ov2722.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov2722.c b/drivers/external_drivers/camera/drivers/media/i2c/ov2722.c
index 486557c..9cf1e4c 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov2722.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov2722.c
@@ -293,7 +293,7 @@ static int ov2722_get_intg_factor(struct i2c_client *client,
 				const struct ov2722_resolution *res)
 {
 	struct v4l2_subdev *sd = i2c_get_clientdata(client);
-	struct ov2722_device *dev = to_ov2722_sensor(sd);
+	struct ov2722_device *dev = NULL;
 	struct atomisp_sensor_mode_data *buf = &info->data;
 	const unsigned int ext_clk_freq_hz = 19200000;
 	const unsigned int pll_invariant_div = 10;
@@ -307,6 +307,8 @@ static int ov2722_get_intg_factor(struct i2c_client *client,
 	if (info == NULL)
 		return -EINVAL;
 
+	dev = to_ov2722_sensor(sd);
+
 	/* pixel clock calculattion */
 	ret =  ov2722_read_reg(client, OV2722_8BIT,
 				OV2722_SC_CMMN_PLL_CTRL3, &pre_pll_clk_div);
-- 
1.9.1

